lib/tree/tree.go

```diff
@@ -0,0 +1,98 @@
+package tree
+
+import (
+	"github.com/dedis/cothority/lib/dbg"
+)
+
+/*
+The tree library holds all hosts and allows for different ways to
+retrieve the hosts - either in a n-ary tree, broadcast or just
+ a selection.
+*/
+
+// Tree holds the structure about the current tree
+type Tree struct {
```

Comment:
You have to convince me first that protobuf is really a good idea ;) For the moment I prefer JSON - it also covers the `map[string]`-type, which is not covered by protobuf, but which I'd like to have to transfer the messages between the peers, so that each round-type can transport it's own messages.

I switched between different tree-structures and settled on this one as it is the most convenient to use in the code. It will probably be marsahlled into a list of indexes to the PeerList, as you suggested.

And if I understood the protobuf-implementation correctly, it doesn't work with different versions of structures at the sending and receiving end. But I might have misread the code.